### PR TITLE
feat(compaction): set file size of sst for different level

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

## What's changed and what's your intention?

close https://github.com/singularity-data/risingwave/issues/3186

- set file size of sst for different level so that we can reduce write-amplification

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by Little-Wallace** [comment]
image
left is master and right is current PR

File: src/meta/src/hummock/compaction/level_selector.rs [hunk 1]
```diff
@@ -220,6 +220,20 @@ impl DynamicLevelSelector {
         ctx.score_levels.sort_by(|a, b| b.0.cmp(&a.0));
         ctx
     }
+
+    fn adjust_target_file_size(&self, mut ret: SearchResult, base_level: usize) -> SearchResult {
```
Comment (Li0k) [inline src/meta/src/hummock/compaction/level_selector.rs:224]: 1. i think need more notes for this algorithm (e.g. simply explain about reduce write amplification)
2. IMO, the parameter name call `ret` is not to make sense

Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/level_selector.rs:224]: How about renaming `SearchResult` to `CompactionInput` or `CompactionContext`?

**Comment by Little-Wallace** [comment]
merge this PR after https://github.com/singularity-data/risingwave/pull/3388

**Pull request merged**